include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -172,94 +186,51 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 				state->layerCount = copyLayerCount;
 			};
 
-			core::vector3du32_SIMD localCoord;
-			for (auto& w = localCoord[3] = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
+			for (uint16_t w = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
```

Comment:
this seems very off, should be `w<copyLayerCount;`

Reply:
the loop is only needed for

```C++
++state->inBaseLayer;
++state->outBaseLayer;
```

what if in/out base layers start with 3 and there is 7 array layers at all?
then the final inBaseLayer will be beyond the available layers in certain loop iteration, won't it?

Reply:
if your layer count + offsets in the original state go out of bounds `validate` should have caught that.

Reply:
your loop is definitely wrong

Reply:
okay, I will check it and correct it

Reply:
corrected